modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectLayoutLocalServiceImpl.java

```diff
@@ -233,8 +231,9 @@ public ObjectLayout updateObjectLayout(
 	}
 
 	private ObjectLayoutBox _addObjectLayoutBox(
-			User user, long objectDefinitionId, long objectLayoutTabId,
-			boolean collapsable, Map<Locale, String> nameMap, int priority,
+			User user, long objectDefinitionId, String objectLayoutBoxType,
```

Comment:
Hi @seltong, just pass objectLayoutBox as parameter and get its propeties.

for example:


			long objectDefinitionId, ObjectLayoutBox objectLayoutBox,
			long objectLayoutTabId, User user)
		throws PortalException {

		ObjectLayoutBox newObjectLayoutBox = _objectLayoutBoxPersistence.create(
			counterLocalService.increment());

		newObjectLayoutBox.setCompanyId(user.getCompanyId());
		newObjectLayoutBox.setUserId(user.getUserId());
		newObjectLayoutBox.setUserName(user.getFullName());
		newObjectLayoutBox.setObjectLayoutTabId(objectLayoutTabId);
		newObjectLayoutBox.setCollapsable(objectLayoutBox.isCollapsable());
		newObjectLayoutBox.setNameMap(objectLayoutBox.getNameMap());
		newObjectLayoutBox.setPriority(objectLayoutBox.getPriority());
            newObjectLayoutBox.setType(objectLayoutBox.getType());

		newObjectLayoutBox =
			_objectLayoutBoxPersistence.update(newObjectLayoutBox);

		newObjectLayoutBox.setObjectLayoutRows(
			_addObjectLayoutRows(
				user, objectDefinitionId,
				newObjectLayoutBox.getObjectLayoutBoxId(),
				objectLayoutBox.getObjectLayoutRows()));

		return newObjectLayoutBox;
	}

Thanks!